kedro/framework/cli/cli.py

```diff
@@ -199,6 +213,19 @@ def main(
                 click.echo(message)
                 click.echo(hint)
             sys.exit(exc.code)
+        except Exception as error:
+            logger.error(f"An error has occurred: {error}")
+            exit_code = 1
+            self._cli_hook_manager.hook.after_command_run(
+                project_metadata=self._metadata, command_args=args, exit_code=1
+            )
+            hook_called = True
+        finally:
+            if not hook_called:
+                self._cli_hook_manager.hook.after_command_run(
+                    project_metadata=self._metadata, command_args=args, exit_code=0
+                )
+            sys.exit(exit_code)
```

Comment:
Is this `sys.exit` necessary? I think there is a high chance this will cause problem to use kedro CLI as `sys.exit` would terminate the program eagerly.

https://github.com/kedro-org/kedro/pull/4026 is trying to change `click` to `standalone=False` to make CLI

https://stackoverflow.com/questions/60780051/python-click-command-exit-flow

This [issue](https://github.com/kedro-org/kedro/issues/3237) document the previous investigation done by @antonymilne and explains more in depth.

Reply:
I've used it here as a way to ensure that it exits with the appropriate error code, because it was exiting with a 0 even when an error happened. It's not necessary to exit at this point, as long as when it exits, it's with the correct code.

Reply:
What I did for now moving the `sys.exit` into the block that handles the exception in case something goes wrong. So if no errors happen, click will exit normally.